src/main/java/com/team4/deskbookingappbe/controller/DeskAvailabilityController.java

```diff
@@ -0,0 +1,46 @@
+package com.team4.deskbookingappbe.controller;
+
+import com.team4.deskbookingappbe.model.domain.DtoDesk;
+import com.team4.deskbookingappbe.model.domain.DtoReservations;
+import com.team4.deskbookingappbe.repository.DeskRepository;
+import com.team4.deskbookingappbe.repository.ReservationsRepository;
+import com.team4.deskbookingappbe.service.ReservationsDataService;
+import io.swagger.v3.oas.annotations.Operation;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping(path = "api/v1")
+public class DeskAvailabilityController {
+
+    private final ReservationsRepository reservationsRepository;
+    private final DeskRepository deskRepository;
+
+    public DeskAvailabilityController(ReservationsRepository reservationsRepository, ReservationsDataService reservationsDataService, DeskRepository deskRepository) {
+        this.reservationsRepository = reservationsRepository;
+        this.deskRepository = deskRepository;
+    }
+
+    @GetMapping(path = "/available")
```

Comment:
```suggestion
    @GetMapping(path = "desks/available")
```
Maybe something like this. available is way too generic. The one above is not perfect either